File: src/main/java/de/pflugradts/passbird/adapter/passwordstore/PasswordStoreFacade.java
package de.pflugradts.passbird.adapter.passwordstore;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import de.pflugradts.passbird.domain.model.password.PasswordEntry;
import de.pflugradts.passbird.domain.service.password.storage.PasswordStoreAdapterPort;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.util.function.Supplier;
import java.util.stream.Stream;

@NoArgsConstructor
@AllArgsConstructor
@Singleton
public class PasswordStoreFacade implements PasswordStoreAdapterPort {

    @Inject
    private PasswordStoreReader passwordStoreReader;
    @Inject
    private PasswordStoreWriter passwordStoreWriter;

    @Override
    public Supplier<Stream<PasswordEntry>> restore() {
        return passwordStoreReader.restore();
    }

    @Override
    public void sync(final Supplier<Stream<PasswordEntry>> passwordEntriesSupplier) {
        passwordStoreWriter.sync(passwordEntriesSupplier);
    }

}

File: src/main/kotlin/de/pflugradts/passbird/adapter/passwordstore/PasswordStoreFacade.kt
package de.pflugradts.passbird.adapter.passwordstore

import com.google.inject.Inject
import com.google.inject.Singleton
import de.pflugradts.passbird.domain.service.password.storage.EggStreamSupplier
import de.pflugradts.passbird.domain.service.password.storage.PasswordStoreAdapterPort

@Singleton
class PasswordStoreFacade @Inject constructor(
    @Inject val passwordStoreReader: PasswordStoreReader,
    @Inject val passwordStoreWriter: PasswordStoreWriter,
) : PasswordStoreAdapterPort {
    override fun restore(): EggStreamSupplier = passwordStoreReader.restore()
    override fun sync(supplier: EggStreamSupplier) { passwordStoreWriter.sync(supplier) }
}
